File: 69. Sqrt(x)/Solution.cpp
class Solution {
public:
    int mySqrt(int x) {
        int right = x; 
        int left = 0;
        int result = 0;
        while(left <= right) {
            int mid = left + (right - left) / 2;
            long long square = (long long)mid * mid;
            if(square == x) {
                return mid;
            }
            if(square < x) {
                result = mid;
                left = mid+1;
            }
            if(square > x) right = mid -1;
        }
        return result;
    }
};

File: 69. Sqrt(x)/Solution.py
class Solution:
    def mySqrt(self, x: int) -> int:
        low, high = 0, x
        while low <= high:
            mid = (low+high)//2
            if mid*mid == x or (mid*mid < x and (mid+1) * (mid+1) > x):
                return mid
            elif mid*mid < x:
                low = mid + 1
            else:
                high = mid - 1